crates/primitives/src/macros.rs

```diff
@@ -33,27 +33,28 @@ macro_rules! impl_buf_wrapper {
     };
 }
 
-pub mod internal {
+pub(crate) mod internal {
     // Crate-internal impls.
 
     macro_rules! impl_buf_common {
         ($name:ident, $len:expr) => {
             impl $name {
                 pub const LEN: usize = $len;
 
+                #[allow(clippy::missing_const_for_fn)]
                 pub fn new(data: [u8; $len]) -> Self {
                     Self(data)
                 }
 
-                pub fn as_slice(&self) -> &[u8] {
+                pub const fn as_slice(&self) -> &[u8] {
                     &self.0
                 }
 
-                pub fn as_mut_slice(&mut self) -> &mut [u8] {
+                pub const fn as_mut_slice(&mut self) -> &mut [u8] {
                     &mut self.0
                 }
 
-                pub fn as_bytes(&self) -> &[u8] {
+                pub const fn as_bytes(&self) -> &[u8] {
                     self.0.as_slice()
                 }
```

Comment:
These *do* make sense to be const imo since they're more general accessors on a general type, which acts like a wrapper around an internal type.